### ISSUE TITLE
"user is already subscribed to the list" notification when adding subgroups/interest groups to contact  

### ISSUE THREAD

**Issue by jason-vivid** [veda-consulting/uk.co.vedaconsulting.mailchimp#139]
This is very minor, but when you add a contact record to a child-group corresponding to an interest group in Mailchimp you get the error in quotes above.  

What this tells me is that when you add the interest groups it is also trying to add the user to the list in mailchimp first. 

Maybe a simple check first to see if they are on the list before you try to add them to the list would clear it up.  If they are on the list already then just check the interest group, otherwise add both.

**Comment by jason-vivid** [comment]
I didn't catch this the  first time around but it seems like the interest groupings (child groups in civi which we have matched up to the mailchimp interst groups) aren't getting auto synchronized when you add them.  I'm not sure if that was the intent or not.  When you run the sync the interest groups get matched up so this isn't the case.  

But for instance I add the sub-group and the warning fires:
  "test1425@test.com is already subscribed to the list."
and then you get the notification:
   "Contact has been added to '*Interest Group Name*'"

We run a sync before doing mailings so it doesn't matter or not, but if the warning is preventing the groups from automatically syncing up people should know that is the intent.
